connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Cast.java

```diff
@@ -0,0 +1,363 @@
+import org.apache.kafka.connect.transforms.util.SchemaUtil;
+import org.apache.kafka.connect.transforms.util.SimpleConfig;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.kafka.connect.transforms.util.Requirements.requireMap;
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+public abstract class Cast<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    // TODO: Currently we only support top-level field casting. Ideally we could use a dotted notation in the spec to
+    // allow casting nested fields.
+    public static final String OVERVIEW_DOC =
+            "Cast fields to a specific type, e.g. to force an integer field to a smaller width. Only simple primitive "
+                    + "types are supported -- integers, floats, boolean, and string. "
+                    + "<p/>Use the concrete transformation type designed for the record key (<code>" + Key.class.getCanonicalName() + "</code>) "
+                    + "or value (<code>" + Value.class.getCanonicalName() + "</code>).";
+
+    public static final String SPEC_CONFIG = "spec";
+
+    public static final ConfigDef CONFIG_DEF = new ConfigDef()
+            .define(SPEC_CONFIG, ConfigDef.Type.LIST, ConfigDef.NO_DEFAULT_VALUE, new ConfigDef.Validator() {
+                    @Override
+                    public void ensureValid(String name, Object valueObject) {
+                        List<String> value = (List<String>) valueObject;
+                        if (value.isEmpty()) {
+                            throw new ConfigException("Must specify at least one field to cast.");
+                        }
+                        parseFieldTypes(value);
+                    }
+
+                    @Override
+                    public String toString() {
+                        return "list of colon-delimited pairs, e.g. <code>foo:bar,abc:xyz</code>";
+                    }
+            },
+            ConfigDef.Importance.HIGH,
+            "List of fields and the type to cast them to of the form field1:type,field2:type to cast fields of "
+                    + "Maps or Structs. A single type to cast the entire value.");
+
+    private static final String PURPOSE = "cast types";
+
+    private static final Set<Schema.Type> SUPPORTED_CAST_TARGETS = new HashSet<>(
+            Arrays.asList(Schema.Type.INT8, Schema.Type.INT16, Schema.Type.INT32, Schema.Type.INT64,
+                    Schema.Type.FLOAT32, Schema.Type.FLOAT64, Schema.Type.BOOLEAN, Schema.Type.STRING)
+    );
+
+    // As a special case for casting the entire value (e.g. the incoming key is a int64 but you know it could be an
+    // int32 and want the smaller width), we use an otherwise invalid field name in the cast spec to track this.
+    private static final String WHOLE_VALUE_CAST = "";
```

Comment:
No, it's not identity. The original idea for the transformation was casting the type of fields (in a Struct w/ schemas, in a Map w/o schemas). But then I realized that if you apply it to keys, it might need to apply to the entire record (either key or value in the ConnectRecord) rather than a field. I was definitely stretching to find a name to refer to this...

Relatedly, there's some future improvements to be added here. Currently the code only handles casting the entire value or fields within the value if it is a Struct/Map. Still missing is handling hierarchical structures, i.e. in the spec, a field would ideally support some sort of dotted notation to allow for casting in nested structs.